### PR TITLE
[FIX] im_livechat: only return active chats from init messaging 

### PR DESCRIPTION
**Pull request by tsm-odoo**
Since [1], only active live chat channels are returned by the
init_messaging rpc since the inactive ones are not required.

This is done by passing the `is_for_livechat` key in the context. Since
[2], the context is not propagated anymore thus inactive channels are
returned as well.

Actually, we shouldn't have to override the `_get_init_channels`
method: the mail one only returns channels that have an opened chat
window. Closing the live chat session should close the existing chat
window as well.

This PR ensures the chat windows will be closed when the visitor
leaves the session so that those outdated channels are not returned by
the initialization of messaging and provides a way to restrict channels
by channel type when initializing messaging instead of relying on the 
context.

[1]: https://github.com/odoo/odoo/pull/146098
[2]: https://github.com/odoo/odoo/pull/150367

enterprise: https://github.com/odoo/enterprise/pull/55490

### CODE REVIEW

File: addons/im_livechat/controllers/main.py [hunk 1]
```diff
@@ -262,6 +262,7 @@ def visitor_leave_session(self, uuid):
             return
         domain = [("channel_id", "=", channel_sudo.id), ("is_self", "=", True)]
         member = request.env["discuss.channel.member"].search(domain)
+        member.fold_state = "closed"
```
Comment (seb-odoo) [inline addons/im_livechat/controllers/main.py:265]: in `_close_livechat_session`?

Comment (seb-odoo) [reply addons/im_livechat/controllers/main.py:265]: all of this method, rtc leave too?

File: addons/mail/controllers/webclient.py [hunk 2]
```diff
@@ -37,16 +39,30 @@ def _process_request(self, **kwargs):
         return store.get_result()
 
     def _process_request_for_all(self, store, **kwargs):
-        if kwargs.get("init_messaging"):
-            if not request.env.user._is_public():
-                user = request.env.user.sudo(False)
-                user._init_messaging(store)
+        if "init_messaging" not in kwargs:
+            return
```
Comment (seb-odoo) [inline addons/mail/controllers/webclient.py:43]: There might/will be more options inside this method, a quick return is not something we'll have in future PR.

Comment (tsm-odoo) [reply addons/mail/controllers/webclient.py]: reverted

**Review by seb-odoo — APPROVED** [review]
@robodoo r+ delegate+

**Comment by tsm-odoo** [comment]
@robodoo r+

Fixed typo in the commit message.

**Pull request closed without merging**